### PR TITLE
DHCP fingerprints for Aruba sensors

### PR DESCRIPTION
**Pull request by dmoinescu-r7**
## Description
New DHCP fingerprints for Aruba sensors, based on data observed inside DHCP packets.

## How Has This Been Tested?
Ran `rake tests`, `recog_verify`, `recog_standardize` and `update_cpes.py`

## Types of changes

- New feature (non-breaking change which adds functionality)

## Checklist:

- [x] I have updated the documentation accordingly (or changes are not required).
- [x] I have added tests to cover my changes (or new tests are not required).
- [x] All new and existing tests passed.

### CODE REVIEW

File: xml/dhcp_vendor_class.xml [hunk 1]
```diff
@@ -190,6 +190,18 @@
     <param pos="0" name="hw.device" value="Switch"/>
   </fingerprint>
 
+  <fingerprint pattern="^Aruba-Sensor(?:-([A-Z0-9]{6}))?$">
```
Comment (mkienow-r7) [inline xml/dhcp_vendor_class.xml:193]: Is `Aruba-Sensor` a valid value? If so, we should probably have an example for it.
```
$ echo "Aruba-Sensor" | bundle exec ./bin/recog_match xml/dhcp_vendor_class.xml
MATCH: {"matched"=>"HP Aruba UXI Sensor", "hw.vendor"=>"Aruba Networks", "hw.model"=>nil, "hw.device"=>"Sensor", "os.vendor"=>"Aruba Networks", "service.protocol"=>"dhcp", "fingerprint_db"=>"dhcp_vendor_class", "data"=>"Aruba-Sensor"}
```

File: xml/dhcp_vendor_class.xml [hunk 2]
```diff
@@ -190,6 +190,18 @@
...
+    <description>HP Aruba UXI Sensor</description>
+    <example hw.model="Q9X65A">Aruba-Sensor-Q9X65A</example>
+    <example hw.model="R3R67A">Aruba-Sensor-R3R67A</example>
+    <example hw.model="R7H75A">Aruba-Sensor-R7H75A</example>
+    <example>Aruba-Sensor</example>
+    <param pos="0" name="hw.vendor" value="Aruba Networks"/>
+    <param pos="1" name="hw.model"/>
+    <param pos="0" name="hw.device" value="Sensor"/>
```
Comment (mkienow-r7) [inline xml/dhcp_vendor_class.xml:201]: Would "Network Management Device" be a better classification than "Sensor"? They are mentioned with the text "network management device" on CDW's website, but I was unable to locate such language on Aruba's website.

File: xml/dhcp_vendor_class.xml (hunk 1 above)
Comment (dmoinescu-r7) [reply xml/dhcp_vendor_class.xml:193]: Yes, it's valid. See the 4th example below (line 198 in the file).

File: xml/dhcp_vendor_class.xml (hunk 2 above)
Comment (dmoinescu-r7) [reply xml/dhcp_vendor_class.xml:201]: I took that text from the [Aruba site](https://www.arubanetworks.com/products/network-management-operations/analytics-monitoring/user-experience-insight-sensors/). Other sites (resellers, presumably) have different descriptions - for instance, model Q9X65A is described in various places as:
* LTE Sensor
* LTE Sensor - wireless network management device
* LTE Wireless Sensor Network Management Device
* User Experience Insight LTE Sensor
* UXI sensor

The full description for the other 2 models mentioned in examples (R3R67A, R7H75A) seems to be _User Experience Insight Sensor G-Series - wireless network management device_.

So I think both `sensor` and `network management device` are reasonable descriptions. I don't have a strong opinion on the matter.

File: xml/dhcp_vendor_class.xml (hunk 1 above)
Comment (mkienow-r7) [reply xml/dhcp_vendor_class.xml:193]: I completely overlooked that example! 😅

File: xml/dhcp_vendor_class.xml (hunk 2 above)
Comment (mkienow-r7) [reply xml/dhcp_vendor_class.xml:201]: To get another opinion, I contacted @TomSellers who agreed both are correct, but "Network Management Device" would be more specific.

Comment (dmoinescu-r7) [reply xml/dhcp_vendor_class.xml]: Updated `hw.device` to "Network Management Device"

**Review by mkienow-r7 — APPROVED** [review]
Thank you for the contribution @dmoinescu-r7!

**Pull request merged**